src/components/DocumentRenderer/QrCode/QrCodeContainer.tsx

```diff
@@ -0,0 +1,163 @@
+import React, {
+  useState,
+  useImperativeHandle,
+  forwardRef,
+  useEffect,
+  useCallback
+} from "react";
+import { StyleSheet, View, Text } from "react-native";
+import {
+  color,
+  borderRadius,
+  shadow,
+  size,
+  fontSize
+} from "../../../common/styles";
+import { QrCode } from "./QrCode";
+import { DocumentProperties, DocumentObject } from "../../../types";
+import { useQrGenerator } from "../../../common/hooks/useQrGenerator";
+import { useNetworkContext } from "../../../context/network";
+import HumanizeDuration from "humanize-duration";
+import { useCountdown } from "../../../common/hooks/useCountdown";
+
+const styles = StyleSheet.create({
+  wrapper: {
+    backgroundColor: color("grey", 0),
+    borderRadius: borderRadius(2),
+    ...shadow(2)
+  },
+  qrCodeWrapper: {
+    aspectRatio: 1
+  },
+  feedback: {
+    marginTop: -size(2)
+  },
+  expiryText: {
+    fontSize: fontSize(-3),
+    marginHorizontal: size(3),
+    marginBottom: size(2),
+    textAlign: "center"
+  },
+  offlineBanner: {
+    backgroundColor: color("red", 10),
+    paddingVertical: size(2),
+    paddingHorizontal: size(3),
+    borderRadius: borderRadius(2),
+    borderTopLeftRadius: 0,
+    borderTopRightRadius: 0
+  },
+  offlineBannerHeader: {
+    fontSize: fontSize(-2),
+    color: color("red", 30),
+    fontWeight: "bold",
+    marginBottom: size(0.5)
+  },
+  offlineBannerText: {
+    fontSize: fontSize(-3),
+    color: color("red", 30)
+  }
+});
+
+const updateDocumentQr = async (
+  qrCode: { url: string; expiry?: number },
+  document: { atomicUpdate: DocumentObject["atomicUpdate"] }
+): Promise<void> => {
+  const updateFunction = (oldData: DocumentProperties): DocumentProperties => {
```

Comment:
Are qrCode linked to the documents? Don’t see some kind of unique identifier to associate them, I might be missing something.